### DISCUSSION TITLE
Alveo U50 in Vivavo 2023.1

### DISCUSSION THREAD

**Discussion by wnew** [Xilinx/open-nic-shell#46] [Q&A]
I have built the AU50 design without any user design and am getting intra-clock timing issues within the CMAC tx clock domain. Is to be expected without user logic?

I have however managed to get it to make timing with synthesis strategy of FlowAreaOptimized_high and implementation strategy congestion_Spreadlogic_high.  

Any suggestions here?

Regards

**Comment by cneely-amd** [comment]
Hi @wnew, I'll just quickly comment that the implementation settings are the ones that would have more timing impact rather than the synthesis settings.  For example, I can imagine that setting to one of the PerformanceExplore ... or Performance related ones with high efforts might give the best timing results.  Yes, I think for some of the boards that it is necessary to go and adjust/manually modify the implementation strategy in order to meet timing.

Thanks,
--Chris

**Reply by wnew** [reply]
Thanks for your reply. I have tested the build that I got through and it all seems to work well.

What would be required for the project to support Vivado 2023.1, is there any plan to add this support in the future?

-Wes

**Comment by cneely-amd** [comment]
Thanks, I'll get back to you after I try the pull request.  

Unfortunately, I have a message to convey regarding Vivado versions below.

In terms of Vivado support, right now v2022.1 is the best version to use, and the reason for this is the QDMA IP.  In Vivado v2022.2 there was a new major release of QDMA (v5.0), it changed its register interface for memory-mapped control registers to support wider values for certain fields.  Effectively, the QDMA team reorganized the register mapping. (In doing so that broke compatibility with the earlier versions of any QDMA-related software).  

open-nic-shell and open-nic-driver were written against earlier versions of QDMA.   As it stands currently, open-nic-driver hasn't been updated for QDMA v5.0.  However, the official Xilinx QDMA driver repo on github contains updates for QDMA v5.0.  So, depending on what you need on the software side, you might see DPDK work, but the separate Linux kernel driver can have issues.  More specifically, open-nic-driver can stop responding at high throughput due to the differences in QDMA v5.0 register interface, but still works properly with QDMA v4.0.

For now, it's probably easiest to stick to using Vivado 2022.1 or earlier.  Or if necessary to generate the QDMA IP using an earlier version, and then not to upgrade that indivdual IP, when switching/using the design with newer versions of Vivado.

This is a rather long message, but hopefully I've explained this issue clearly.

**Comment by vaniaprkl** [comment]
Hi @cneely-amd, I am running to a very frustrating issue with 2022.1 version of Vivado. I chose it specifically based on your recommendation above to @wnew, who referred me to this thread. 

 I have installed the Vivado Standard edition and when I build opennic-shell it gets stuck on not being able to generate clk IP.
 
 ```
 wait_on_runs: Time (s): cpu = 00:00:53 ; elapsed = 00:01:00 . Memory (MB): peak = 3075.641 ; gain = 0.000 ; free physical = 14641 ; free virtual = 252223
WARNING: [Vivado 12-818] No files matched '/disk111/opennic/open-nic-shell/build/au50_u50buildAug31/vivado_ip/clk_wiz_50Mhz/clk_wiz_50Mhz.xci'
WARNING: [Vivado 12-818] No files matched '/disk111/opennic/open-nic-shell/build/au50_u50buildAug31/vivado_ip/clk_wiz_50Mhz/clk_wiz_50Mhz.xci'
expected floating-point number but got "Unable to get value from speedsfile for keyword MM"
ERROR: [IP_Flow 19-3476] Tcl error in create_gui procedure for IP 'clk_wiz_50Mhz'. expected floating-point number but got "Unable to get value from speedsfile for keyword MM"
ERROR: [IP_Flow 19-3428] Failed to create Customization object clk_wiz_50Mhz
CRITICAL WARNING: [IP_Flow 19-5622] Failed to create IP instance 'clk_wiz_50Mhz'. Failed to customize IP instance 'clk_wiz_50Mhz'. Failed to load customization data
ERROR: [Common 17-69] Command failed: Create IP failed with errors

    while executing
"source ${ip_tcl_dir}/${ip}.tcl"
    invoked from within
"if {[file exists "${ip_tcl_dir}/${ip}_${board}.tcl"]} {
            source ${ip_tcl_dir}/${ip}_${board}.tcl
        } elseif {[file exists "${ip_tcl_d..."
    ("foreach" body line 30)
    invoked from within
"foreach ip $ips {
        # Pre-save IP name and its build directory to a global dictionary
        dict append ip_dict $ip ${ip_build_dir}/${ip}

   ..."
    ("dict for" body line 17)
    invoked from within
"dict for {module module_dir} $module_dict {
    set ip_tcl_dir ${module_dir}/vivado_ip

    # Check the existence of "$ip_tcl_dir" and "${ip_tcl_dir}/..."
    (file "build.tcl" line 255)
INFO: [Common 17-206] Exiting Vivado at Thu Aug 31 19:07:58 2023...

 ```
 
 
 So I opened Vivado GUI and checked if the clocking wizard can even be opened and I run into this error. This specifically happen when I target opennic shell to U50 or U55C board with this version of Vivado 2022.1. Any suggestions?  
 
![Screenshot 2023-09-01 at 10 15 25 AM](https://github.com/Xilinx/open-nic-shell/assets/104578360/6a24e6c8-7467-4e56-b37c-9d160c95efe4)
